src/main/java/com/mongodb/jdbc/MongoResultSet.java

```diff
@@ -813,14 +814,98 @@ public ResultSetMetaData getMetaData() throws SQLException {
 
     @Override
     public Object getObject(int columnIndex) throws SQLException {
-        throw new SQLFeatureNotSupportedException(
-                Thread.currentThread().getStackTrace()[1].toString());
+		checkBounds(columnIndex);
+		int zeroIndex = columnIndex - 1;
+		BsonValue out = current.getValues().get(columnIndex - 1).value;
+		int columnType = rsMetaData.getColumnType(columnIndex);
+		switch (columnType) {
+			case Types.ARRAY:
+				// not supported
```

Comment:
java switches are like c switches, so this types.ARRAY will fall through to types.INT and then fail with an error that will be strange to users. I suggest putting all the unsupported cases at the end so that it goes to the Exception have you at the bottom. Other option is to put in breaks in all of these, but I think that's more verbose.

Reply:
updated! I was starting out by writing a `throw` in each of these cases, and later decided to just factor it out to the end (but forgot to add the breaks in here)! I'm going to add breaks instead of reordering so that this can be left in alphabetical order